spec/models/reports/patient_state_spec.rb

```diff
@@ -225,6 +244,151 @@
+        patient_1 = create(:patient, recorded_at: june_2021[:long_ago])
+        create(:encounter, patient: patient_1, encountered_on: june_2021[:now] - 3.months)
+        patient_2 = create(:patient, recorded_at: june_2021[:long_ago])
+        create(:encounter, patient: patient_2, encountered_on: june_2021[:now] - 1.month)
+        patient_3 = create(:patient, recorded_at: june_2021[:long_ago])
+        RefreshMaterializedViews.new.refresh_v2
+
+        with_reporting_time_zone do
+          expect(described_class.where(htn_treatment_outcome_in_last_2_months: "missed_visit", month_date: june_2021[:now]).pluck(:patient_id))
+            .to include(patient_1.id, patient_3.id)
+          expect(described_class.where(htn_treatment_outcome_in_last_2_months: "missed_visit", month_date: june_2021[:now]).pluck(:patient_id))
+            .not_to include(patient_2.id)
+        end
+      end
+      it "visited_no_bp if the patient visited, but didn't get a BP taken in the last 2 months" do
+        patient_bp_over_2_months = create(:patient, recorded_at: june_2021[:long_ago])
+        create(:prescription_drug,
+          device_created_at: june_2021[:now] - 1.month,
+          facility: patient_bp_over_2_months.registration_facility,
+          patient: patient_bp_over_2_months,
+          user: patient_bp_over_2_months.registration_user)
+        create(:blood_pressure, patient: patient_bp_over_2_months, recorded_at: june_2021[:now] - 2.months)
+
+        patient_bp_under_2_months = create(:patient, recorded_at: june_2021[:long_ago])
+        create(:prescription_drug,
+          device_created_at: june_2021[:now] - 1.month,
+          facility: patient_bp_under_2_months.registration_facility,
+          patient: patient_bp_under_2_months,
+          user: patient_bp_under_2_months.registration_user)
+        create(:blood_pressure, patient: patient_bp_under_2_months, recorded_at: june_2021[:now] - 1.month)
+
+        patient_with_no_bp = create(:patient, recorded_at: june_2021[:long_ago])
+        create(:prescription_drug,
+          device_created_at: june_2021[:now] - 1.month,
+          facility: patient_with_no_bp.registration_facility,
+          patient: patient_with_no_bp,
+          user: patient_with_no_bp.registration_user)
+        RefreshMaterializedViews.new.refresh_v2
+
+        with_reporting_time_zone do
+          expect(described_class.where(htn_treatment_outcome_in_last_2_months: "visited_no_bp", month_date: june_2021[:now]).pluck(:patient_id))
+            .to include(patient_bp_over_2_months.id, patient_with_no_bp.id)
+          expect(described_class.where(htn_treatment_outcome_in_last_2_months: "visited_no_bp", month_date: june_2021[:now]).pluck(:patient_id))
+            .not_to include(patient_bp_under_2_months.id)
+        end
+      end
+      it "controlled/uncontrolled if there is a BP measured in the last 2 months that is under/not under control" do
```

Comment:
missing some spacing here

Reply:
Also I think it's ok to split this into two examples. 4 outcomes, 4 specs.

Reply:
Fixed spacing in ed649e75367c82231d69e97799b2407b63c4b818 and split examples in c39ffe8d4ee95c936b28fa3ae8d4e168a3cc97e9.